core/templates/components/skill-selector/skill-selector.component.ts

```diff
@@ -0,0 +1,213 @@
+  }
+
+  // The folowing function is called when the subtopic filter changes.
+  // This updates the list of Skills displayed in the selector.
+  updateSkillsListOnSubtopicFilterChange(): void {
+    var updatedSkillsDict = {};
+    var isAnySubTopicChecked = false;
+    for (let topicName in this.subTopicFilterDict) {
+      var subTopics = this.subTopicFilterDict[topicName];
+      for (var i = 0; i < subTopics.length; i++) {
+        if (subTopics[i].checked) {
+          if (!updatedSkillsDict.hasOwnProperty(topicName)) {
+            updatedSkillsDict[topicName] = {};
+          }
+          var tempCategorizedSkills = this.categorizedSkills;
+          var subTopicName = subTopics[i].subTopicName;
+          updatedSkillsDict[topicName][subTopicName] =
+            tempCategorizedSkills[topicName][subTopicName];
+          isAnySubTopicChecked = true;
+        }
+      }
+    }
+    if (!isAnySubTopicChecked) {
+      // If no subtopics are checked in the subtop filter, we have
+      // to display all the skills from checked topics.
+      var isAnyTopicChecked = false;
+      for (var i = 0; i < this.topicFilterList.length; i++) {
+        if (this.topicFilterList[i].checked) {
+          var tempCategorizedSkills = this.categorizedSkills;
+          var topicName: string = this.topicFilterList[i].topicName;
+          updatedSkillsDict[topicName] = tempCategorizedSkills[topicName];
+          isAnyTopicChecked = true;
+        }
+      }
+      if (isAnyTopicChecked) {
+        this.currCategorizedSkills = angular.copy(updatedSkillsDict);
+      } else {
+        // If no filter is applied on both subtopics and topics, we
+        // need to display all the skills (the original list).
+        this.currCategorizedSkills = angular.copy(this.categorizedSkills);
+      }
+    } else {
+      this.currCategorizedSkills = angular.copy(updatedSkillsDict);
+    }
+  }
+
+  // The folowing function is called when the topic filter changes.
+  // First, the subtopic filter is updated according to the changed
+  // topic filter list. Then the main Skills list is updated.
+  updateSkillsListOnTopicFilterChange(): void {
+    var updatedSubTopicFilterList = {};
+    var isAnyTopicChecked = false;
+    for (var i = 0; i < this.topicFilterList.length; i++) {
+      if (this.topicFilterList[i].checked) {
+        var topicName = this.topicFilterList[i].topicName;
+        updatedSubTopicFilterList[topicName] = (
+          angular.copy(this.intialSubTopicFilterDict[topicName]));
+        isAnyTopicChecked = true;
+      }
+    }
+    if (!isAnyTopicChecked) {
+      // If there are no topics checked on topic filter, we have to
+      // display subtopics from all the topics in the subtopic filter.
+      for (var topic in this.intialSubTopicFilterDict) {
+        if (!this.subTopicFilterDict.hasOwnProperty(topic)) {
+          this.subTopicFilterDict[topic] = (
+            angular.copy(this.intialSubTopicFilterDict[topic]));
```

Comment:
angular.copy() is only available in angularjs.

Reply:
Done